File: Go/01.Go-Basic-February-2022/06.conditionalStatementAdvanced-Exercise/03.newHouse.go
package main

import "fmt"

func main() {
	var typeFlower string
	fmt.Scanln(&typeFlower)

	var count int
	fmt.Scanln(&count)

	var budget int
	fmt.Scanln(&budget)

	var price float32

	switch typeFlower {
	case "Roses":
		price = 5
	case "Dahlias":
		price = 3.8
	case "Tulips":
		price = 2.8
	case "Narcissus":
		price = 3
	case "Gladiolus":
		price = 2.5
	}

	var discount float32 = 0

	if typeFlower == "Roses" && count > 80 {
		discount = 0.10
	} else if typeFlower == "Dahlias" && count > 90 {
		discount = 0.15
	} else if typeFlower == "Tulips" && count > 80 {
		discount = 0.15
	} else if typeFlower == "Narcissus" && count < 120 {
		price = price * 1.15
	} else if typeFlower == "Gladiolus" && count < 80 {
		price = price * 1.2
	}

	var total float32 = float32(count) * (1 - discount) * price

	if float32(budget) >= total {
		var rest float32 = float32(budget) - total
		fmt.Printf("Hey, you have a great garden with %d %s and %.2f leva left.", count, typeFlower, rest)
	} else {
		fmt.Printf("Not enough money, you need %.2f leva more.", total-float32(budget))
	}
}

File: JavaScript/01.JavaScript-Basic-February-2021/06.conditionalStatementsAdvanced-Exercise/03.newHouse.js
function newHouse(input) {
    let type = input[0];
    let count = Number(input[1]);
    let budget = Number(input[2]);

    let money = 0;
    switch (type) {
        case "Roses":
            money = count * 5;
            if (count > 80) {
                money = money * 0.9;
            }
            break;

        case "Dahlias":
            money = count * 3.8;
            if (count > 90) {
                money = money * 0.85;
            }
            break;

        case "Tulips":
            money = count * 2.8;
            if (count > 80) {
                money = money * 0.85;
            }
            break;

        case "Narcissus":
            money = count * 3;
            if (count < 120) {
                money = money * 1.15;
            }
            break;

        case "Gladiolus":
            money = count * 2.5;
            if (count < 80) {
                money = money * 1.2;
            }
            break;

        default:
            break;
    }

    if (budget >= money) {
        console.log(`Hey, you have a great garden with ${count} ${type} and ${(budget - money).toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money, you need ${(money - budget).toFixed(2)} leva more.`);
    }
}

newHouse(["Roses", "55", "250"]);
newHouse(["Tulips", "88", "260"]);